File: runtime/dart/typed_data/unmodifiableByteBufferView.ts
import { IByteBuffer } from "./byteBuffer";
import { IByteData } from "./byteData";
import { IFloat32List } from "./float32List";
import { IFloat32x4List } from "./float32x4List";
import { IFloat64List } from "./float64List";
import { IFloat64x2List } from "./float64x2List";
import { IInt16List } from "./int16List";
import { IInt32List } from "./int32List";
import { IInt32x4List } from "./int32x4List";
import { IInt64List } from "./int64List";
import { IInt8List } from "./int8List";
import { IUint16List } from "./uint16List";
import { IUint32List } from "./uint32List";
import { IUint64List } from "./uint64List";
import { IUint8ClampedList } from "./uint8ClampedList";
import { IUint8List } from "./uint8List";
declare const dart: {
    typed_data: {
        unmodifiableByteBufferView: (
            this: void,
            unmodifiableByteBufferView: IUnmodifiableByteBufferView,
            data: IByteBuffer
        ) => IUnmodifiableByteBufferView;
    };
};
export interface IUnmodifiableByteBufferView {
    getLengthInBytes: () => number;
    asUint8List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint8List;
    asInt8List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt8List;
    asUint8ClampedList: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint8ClampedList;
    asUint16List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint16List;
    asInt16List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt16List;
    asUint32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint32List;
    asInt32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt32List;
    asUint64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint64List;
    asInt64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt64List;
    asInt32x4List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt32x4List;
    asFloat32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat32List;
    asFloat64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat64List;
    asFloat32x4List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat32x4List;
    asFloat64x2List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat64x2List;
    asByteData: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IByteData;
    toString: () => string;
    getHashCode: () => number;
}
export class UnmodifiableByteBufferView implements IByteBuffer {
    public constructor(data: IByteBuffer) {
        dart.typed_data.unmodifiableByteBufferView(this, data);
    }
    private readonly _dart_getLengthInBytes: () => number = undefined as any;
    private readonly _dart_asUint8List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint8List = undefined as any;
    private readonly _dart_asInt8List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt8List = undefined as any;
    private readonly _dart_asUint8ClampedList: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint8ClampedList = undefined as any;
    private readonly _dart_asUint16List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint16List = undefined as any;
    private readonly _dart_asInt16List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt16List = undefined as any;
    private readonly _dart_asUint32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint32List = undefined as any;
    private readonly _dart_asInt32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt32List = undefined as any;
    private readonly _dart_asUint64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IUint64List = undefined as any;
    private readonly _dart_asInt64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt64List = undefined as any;
    private readonly _dart_asInt32x4List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IInt32x4List = undefined as any;
    private readonly _dart_asFloat32List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat32List = undefined as any;
    private readonly _dart_asFloat64List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat64List = undefined as any;
    private readonly _dart_asFloat32x4List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat32x4List = undefined as any;
    private readonly _dart_asFloat64x2List: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IFloat64x2List = undefined as any;
    private readonly _dart_asByteData: (
        offsetInBytes: number,
        length?: number | undefined
    ) => IByteData = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public getLengthInBytes(): number {
        return this._dart_getLengthInBytes();
    }
    public asUint8List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IUint8List {
        return this._dart_asUint8List(offsetInBytes, length);
    }
    public asInt8List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IInt8List {
        return this._dart_asInt8List(offsetInBytes, length);
    }
    public asUint8ClampedList(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IUint8ClampedList {
        return this._dart_asUint8ClampedList(offsetInBytes, length);
    }
    public asUint16List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IUint16List {
        return this._dart_asUint16List(offsetInBytes, length);
    }
    public asInt16List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IInt16List {
        return this._dart_asInt16List(offsetInBytes, length);
    }
    public asUint32List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IUint32List {
        return this._dart_asUint32List(offsetInBytes, length);
    }
    public asInt32List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IInt32List {
        return this._dart_asInt32List(offsetInBytes, length);
    }
    public asUint64List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IUint64List {
        return this._dart_asUint64List(offsetInBytes, length);
    }
    public asInt64List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IInt64List {
        return this._dart_asInt64List(offsetInBytes, length);
    }
    public asInt32x4List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IInt32x4List {
        return this._dart_asInt32x4List(offsetInBytes, length);
    }
    public asFloat32List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IFloat32List {
        return this._dart_asFloat32List(offsetInBytes, length);
    }
    public asFloat64List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IFloat64List {
        return this._dart_asFloat64List(offsetInBytes, length);
    }
    public asFloat32x4List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IFloat32x4List {
        return this._dart_asFloat32x4List(offsetInBytes, length);
    }
    public asFloat64x2List(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IFloat64x2List {
        return this._dart_asFloat64x2List(offsetInBytes, length);
    }
    public asByteData(
        offsetInBytes: number = 0,
        length?: number | undefined
    ): IByteData {
        return this._dart_asByteData(offsetInBytes, length);
    }
    public toString(): string {
        return this._dart_toString();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}

File: lib/cfr/builtins/libs/dart/typed_data/unmodifiableByteBufferView.dart
import 'dart:core';
import 'dart:typed_data';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedUnmodifiableByteBufferView
    extends VMManagedBox<UnmodifiableByteBufferView> {
  VMManagedUnmodifiableByteBufferView(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['getLengthInBytes'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.lengthInBytes,
      ];
    });
    table['asUint8List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint8List>(
            object: vmObject.asUint8List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asInt8List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int8List>(
            object: vmObject.asInt8List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asUint8ClampedList'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint8ClampedList>(
            object: vmObject.asUint8ClampedList(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asUint16List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint16List>(
            object: vmObject.asUint16List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asInt16List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int16List>(
            object: vmObject.asInt16List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asUint32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint32List>(
            object: vmObject.asUint32List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asInt32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int32List>(
            object: vmObject.asInt32List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asUint64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint64List>(
            object: vmObject.asUint64List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asInt64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int64List>(
            object: vmObject.asInt64List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asInt32x4List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int32x4List>(
            object: vmObject.asInt32x4List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asFloat32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float32List>(
            object: vmObject.asFloat32List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asFloat64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float64List>(
            object: vmObject.asFloat64List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asFloat32x4List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float32x4List>(
            object: vmObject.asFloat32x4List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asFloat64x2List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float64x2List>(
            object: vmObject.asFloat64x2List(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['asByteData'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<ByteData>(
            object: vmObject.asByteData(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable()),
      ];
    });
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
    table['getHashCode'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.hashCode,
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final UnmodifiableByteBufferView vmObject;
}

class RTManagedUnmodifiableByteBufferView extends UnmodifiableByteBufferView
    implements Box<UnmodifiableByteBufferView> {
  RTManagedUnmodifiableByteBufferView(ByteBuffer data,
      {required this.table, required this.hydroState})
      : super(
          data,
        ) {
    table['vmObject'] = vmObject;
    table['unwrap'] = makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [unwrap()];
    });
    table['_dart_getLengthInBytes'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.lengthInBytes];
    });
    table['_dart_asUint8List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint8List>(
            object:
                super.asUint8List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asInt8List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int8List>(
            object:
                super.asInt8List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asUint8ClampedList'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint8ClampedList>(
            object: super.asUint8ClampedList(
                luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asUint16List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint16List>(
            object: super
                .asUint16List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asInt16List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int16List>(
            object:
                super.asInt16List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asUint32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint32List>(
            object: super
                .asUint32List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asInt32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int32List>(
            object:
                super.asInt32List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asUint64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Uint64List>(
            object: super
                .asUint64List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asInt64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int64List>(
            object:
                super.asInt64List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asInt32x4List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Int32x4List>(
            object: super
                .asInt32x4List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asFloat32List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float32List>(
            object: super
                .asFloat32List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asFloat64List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float64List>(
            object: super
                .asFloat64List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asFloat32x4List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float32x4List>(
            object: super
                .asFloat32x4List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asFloat64x2List'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<Float64x2List>(
            object: super
                .asFloat64x2List(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_asByteData'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        maybeBoxObject<ByteData>(
            object:
                super.asByteData(luaCallerArguments[1], luaCallerArguments[2]),
            hydroState: hydroState,
            table: HydroTable())
      ];
    });
    table['_dart_toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.toString()];
    });
    table['_dart_getHashCode'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [super.hashCode];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  UnmodifiableByteBufferView unwrap() => this;
  UnmodifiableByteBufferView get vmObject => this;
  @override
  int get lengthInBytes {
    Closure closure = table["getLengthInBytes"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  Uint8List asUint8List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asUint8List"];
    return maybeUnBoxAndBuildArgument<Uint8List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Int8List asInt8List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asInt8List"];
    return maybeUnBoxAndBuildArgument<Int8List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Uint8ClampedList asUint8ClampedList([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asUint8ClampedList"];
    return maybeUnBoxAndBuildArgument<Uint8ClampedList, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Uint16List asUint16List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asUint16List"];
    return maybeUnBoxAndBuildArgument<Uint16List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Int16List asInt16List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asInt16List"];
    return maybeUnBoxAndBuildArgument<Int16List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Uint32List asUint32List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asUint32List"];
    return maybeUnBoxAndBuildArgument<Uint32List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Int32List asInt32List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asInt32List"];
    return maybeUnBoxAndBuildArgument<Int32List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Uint64List asUint64List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asUint64List"];
    return maybeUnBoxAndBuildArgument<Uint64List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Int64List asInt64List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asInt64List"];
    return maybeUnBoxAndBuildArgument<Int64List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Int32x4List asInt32x4List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asInt32x4List"];
    return maybeUnBoxAndBuildArgument<Int32x4List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Float32List asFloat32List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asFloat32List"];
    return maybeUnBoxAndBuildArgument<Float32List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Float64List asFloat64List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asFloat64List"];
    return maybeUnBoxAndBuildArgument<Float64List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Float32x4List asFloat32x4List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asFloat32x4List"];
    return maybeUnBoxAndBuildArgument<Float32x4List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  Float64x2List asFloat64x2List([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asFloat64x2List"];
    return maybeUnBoxAndBuildArgument<Float64x2List, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  ByteData asByteData([int offsetInBytes = 0, int? length]) {
    Closure closure = table["asByteData"];
    return maybeUnBoxAndBuildArgument<ByteData, dynamic>(
        closure.dispatch([table], parentState: hydroState)[0],
        parentState: hydroState);
  }

  @override
  String toString() {
    Closure closure = table["__tostring"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }

  @override
  int get hashCode {
    Closure closure = table["getHashCode"];
    return closure.dispatch([table], parentState: hydroState)[0];
  }
}

void loadUnmodifiableByteBufferView(
    {required HydroState hydroState, required HydroTable table}) {
  table['unmodifiableByteBufferView'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      RTManagedUnmodifiableByteBufferView(
          maybeUnBoxAndBuildArgument<ByteBuffer, dynamic>(luaCallerArguments[1],
              parentState: hydroState),
          table: luaCallerArguments[0],
          hydroState: hydroState)
    ];
  });
  registerBoxer<UnmodifiableByteBufferView>(boxer: (
      {required UnmodifiableByteBufferView vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedUnmodifiableByteBufferView(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}
